python/src/aac/plugins/gen_design_doc/gen_design_doc_impl.py

```diff
@@ -95,14 +91,12 @@ def _get_output_file_extension(template_filespec: str) -> str:
     return extension
 
 
-def _get_from_parsed_models(parsed_models: dict, aac_type: str) -> list:
-    aac_types = [util.get_models_by_type(m, aac_type) for m in parsed_models]
-    return list(flatten([m.values() for m in aac_types]))
+def _get_and_prepare_definitions_by_type(parsed_definitions: Definition, aac_type: str) -> list:
+    filtered_definitions = get_definitions_by_root_key(aac_type, parsed_definitions)
+    return list([definition.structure for definition in filtered_definitions])
```

Comment:
https://github.com/jondavid-black/AaC/pull/248#discussion_r840776253
```suggestion
    return [definition.structure for definition in filtered_definitions]
```